### PR TITLE
Add IoT Security to App/Addon

### PR DESCRIPTION
**Pull request by paulmnguyen**
## Description

IoT Security integration with Palo Alto Networks Splunk Application.

## Motivation and Context

Provide visibility into IoT Security

## How Has This Been Tested?

Tested on Standalone server.

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes if appropriate.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Splunk_TA_paloalto/bin/input_module_iot_security.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        'customerid': opt_customer_id,
+        'key_id': opt_access_key_id,
+        'access_key': opt_secret_access_key,
+        'pagelength': 1000,
+        'offset': 0,
+    }
+
+    last_device_pull = helper.get_check_point("last_run_timestamp")
+
+    if not last_device_pull or datetime.datetime.strptime(last_device_pull, "%Y-%m-%d %H:%M:%S") < datetime.datetime.now() - datetime.timedelta(minutes=5):
```
Comment (btorresgil) [inline Splunk_TA_paloalto/bin/input_module_iot_security.py:124]: Need a comment here about why we chose this criteria for performing a device fetch.

**Review by btorresgil — CHANGES_REQUESTED** [review]
There's a lot of formatting changes in the files that aren't related to IoT. Also a lot of changes to wildfire and autofocus that aren't related. Can these be put in a different PR? Where they made by Add-on Builder?

File: Splunk_TA_paloalto/README.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+This is an add-on powered by the Splunk Add-on Builder.
```
[the remark this answers is not in the document]
Comment (paulmnguyen) [reply Splunk_TA_paloalto/README.txt:1]: Yes

File: Splunk_TA_paloalto/README/inputs.conf.spec [hunk 3]
```diff
@@ -1,19 +1,16 @@
...
-credentials =
+[autofocus_export://<name>]
+label = 
 
-# [threatlist://<name>]
-# description =
-# interval =
-# disabled =
-# type =
-# url =
```
[the remark this answers is not in the document]
Comment (paulmnguyen) [reply Splunk_TA_paloalto/README/inputs.conf.spec:19]: Yea also add on builder

File: Splunk_TA_paloalto/app.manifest [hunk 4]
```diff
@@ -5,7 +5,7 @@
     "id": {
       "group": null,
       "name": "Splunk_TA_paloalto",
-      "version": "6.5.0"
+      "version": "6.6.0b"
```
[the remark this answers is not in the document]
Comment (paulmnguyen) [reply Splunk_TA_paloalto/app.manifest:8]: Sorry i changed it in order to create an export in add on builder

File: Splunk_TA_paloalto/default/app.conf [hunk 5]
```diff
@@ -1,18 +1,20 @@
 [install]
 is_configured = false
+state_change_requires_restart = true
```
[the remark this answers is not in the document]
Comment (paulmnguyen) [reply Splunk_TA_paloalto/default/app.conf:3]: Yup add on builder added it.

File: Splunk_TA_paloalto/bin/input_module_iot_security.py [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+
+def validate_input(helper, definition):
+    """Implement your own validation logic to validate the input stanza configurations"""
+    # This example accesses the modular input variable
+    customer_id = definition.parameters.get('customer_id', None)
+    access_key_id = definition.parameters.get('access_key_id', None)
+    secret_access_key = definition.parameters.get('secret_access_key', None)
+    pass
+
+def query_api(helper, url, parameters, api_type, proxy_enabled):
```
[the remark this answers is not in the document]
Comment (btorresgil) [reply Splunk_TA_paloalto/bin/input_module_iot_security.py:30]: You probably don't have time to do this before you leave, so we'll leave it as-is. But I'd like to write unit tests and refactor this at some point.

File: Splunk_TA_paloalto/app.manifest [hunk 7]
```diff
@@ -19,7 +19,7 @@
     "classification": {
       "intendedAudience": null,
       "categories": [],
-      "developmentStatus": "Production/Stable"
+      "developmentStatus": "Beta"
```
[the remark this answers is not in the document]
Comment (btorresgil) [reply Splunk_TA_paloalto/app.manifest]: Added this to develop

**Review by btorresgil — APPROVED** [review]

**Pull request merged**